parliament_challenge.py

```diff
@@ -0,0 +1,111 @@
+import requests
+import waitress
+import flask
+from flask import Flask, request, json
+
+
+def filter_speeches_dict(speeches):
+    """
+    Filter speeches data for relevant items.
+    """
+
+    speeches_list = []
+    for s in speeches:
+        speechesDict = {
+            "anforande_id": s.get("anforande_id"),
+            "dok_datum": s.get("dok_datum"),
+            "parti": s.get("parti"), 
+            "avsnittsrubrik": s.get("avsnittsrubrik"),
+            "links": [{ 
+                "rel": "speech",
+                "href": s.get("protokoll_url_www")
+                    }],
+            "intressent_id": s.get("intressent_id")
+            }
+        speeches_list.append(speechesDict)
+    return speeches_list
+
+
+def filter_member_dict(member):
+    """
+    Filter member data for relevant items.
+    """
+
+    membersDict = {
+        "tilltalsnamn": member.get("tilltalsnamn"),
+        "valkrets": member.get("valkrets"),
+        "bild_url_192": member.get("bild_url_192"),
+        "intressent_id": member.get("intressent_id")
+    }
+    return membersDict
+
+
+def get_member_data(intressent_id):
+    """
+    Get memeber data filter for relevant items.
+    """
+
+    domain = 'http://data.riksdagen.se'
+    members = 'personlista'
+    format_type = 'json'
+    response = requests.get('{}/{}/?iid={}&utformat={}'.format(domain, members, intressent_id, format_type))
+    if response.status_code == 200:
+        data = response.json()
+        member = data["personlista"]["person"]
+        filtered_member = filter_member_dict(member)
+        return filtered_member
+
+
+def get_speeches_data(anftyp, size):
+    """
+    Get speeches data and filter for relevant items.
+    Loop through filtered speeches to get "intressent id" 
+    and pass it as an argurment to the get_member_data 
+    to get member data.
+    Update the member data with speech data.
+    """
+
+    domain = 'http://data.riksdagen.se'
+    speeches = 'anforandelista'
+    format_type = 'json'
+    response = requests.get('{}/{}/?anftyp={}&sz={}&utformat={}'.format(domain, speeches, anftyp, size, format_type))
+    if response.status_code == 200:
+        data = response.json()
+        speeches = data["anforandelista"]["anforande"]
+        filtered_speeches = filter_speeches_dict(speeches)
+        for speech in filtered_speeches:
+            member_data = get_member_data(speech["intressent_id"])        
+            speech.update(member_data)
+        return filtered_speeches
+    else:
+        return []        
+        
+
+def create_app():
+    app = Flask("Parliament Challenge")
+
+    @app.route('/latest-speeches', methods=['GET'])
+    def get_latest_speeches():
+        """
+        GET request:
+            curl -H "Content-Type: application/json" -X GET -d "{\"anftyp\":\"Nej\", \"size\":10}" "localhost:8080/latest-speeches"
```

Comment:
The `-d` flag refers to sending "data" and the client only sends data in `PUT` and `POST` calls, not `GET`.

For the `GET` REST calls, use the  query parameters. Google on query parameters using flask for REST APIs or see the previous projects how you passed query parameters and processed them!

Reply:
used query parameter.
curl request
>> curl -X GET "localhost:8080/latest-speeches/?anftyp=Nej&sz=2"

to get values from url
anftyp = flask.request.args["anftyp"]
size = flask.request.args["sz"]